buildenv/jenkins/docker-slaves/x86/centos6.9/Dockerfile

```diff
@@ -116,9 +116,12 @@ RUN yum -y update \
   && yum -y install devtoolset-7-gcc-7.3* devtoolset-7-binutils \
   && yum -y install devtoolset-7-gcc-c++-7.3* devtoolset-2-gcc-gfortran-7.3* \
   && yum clean all \
-  && ln -s /opt/rh/devtoolset-7/root/usr/bin/gcc /usr/bin/cc \
-  && ln -s /opt/rh/devtoolset-7/root/usr/bin/gcc /usr/bin/gcc \
-  && ln -s /opt/rh/devtoolset-7/root/usr/bin/g++ /usr/bin/g++
+  && echo "source /opt/rh/devtoolset-7/enable" > /etc/profile.d/gcc-7.sh
+
+# Any commands that require gcc need to be run after this SHELL command and before the next SHELL
+# Additionally any commands that call the USER variable will need to be run before this SHELL command or
+# after the next SHELL command, because the USER variable will be overwritten to root instead of being seen as jenkins
```

Comment:
Rather than a comment about `USER`, I suggest its definition should move after the other `SHELL` definition so it's less likely someone in the future would  introduce an earlier reference.